doc: Mention the missing reflected operations

paulofreitas opened the pull request:
Just a minor doc update to maintain consistency with other reflected operations documented in `reference/expressions`. 👍 

<!-- readthedocs-preview cpython-previews start -->
----
📚 Documentation preview 📚: https://cpython-previews--119931.org.readthedocs.build/

<!-- readthedocs-preview cpython-previews end -->

paulofreitas committed:
Mention `__rsub__` arithmetic operation

To maintain consistency with other arithmetic operations.

Eclips4 reviewed (approved):
Thanks!

mdickinson commented on Doc/reference/expressions.rst:
```diff
@@ -1367,7 +1367,7 @@ This operation can be customized using the special :meth:`~object.__add__` and
 numeric arguments are first converted to a common type.
 
-This operation can be customized using the special :meth:`~object.__sub__` method.
+This operation can be customized using the special :meth:`~object.__sub__` and :meth:`~object.__rsub__` methods.
```
Nit: should we wrap this line? It's a little long.

mdickinson replied:
For convenience, here's the suggested wrapping in the form of an actual suggestion:

```suggestion
This operation can be customized using the special :meth:`~object.__sub__`
and :meth:`~object.__rsub__` methods.
```

serhiy-storchaka committed:
Update Doc/reference/expressions.rst

Co-authored-by: Mark Dickinson <dickinsm@gmail.com>

serhiy-storchaka reviewed (changes requested):
There is also the same issue for `__rpow__`, `__rtruediv__`, `__rfloordiv__`, `__rmod__`, `__rlshift__`, `__rrshift__`, and maybe others.

It is better to fix them all in a single PR.

Alternatively, we can remove references to `__radd__` and `__rmul__`.

paulofreitas committed:
Add the other reflected methods

Add links to `__rpow__`, `__rtruediv__`, `__rfloordiv__`, `__rmod__`, `__rlshift__` and `__rrshift__`.

paulofreitas committed:
Merge branch 'main' into rsub-arithmetic-operation

serhiy-storchaka reviewed (commented):
For divisions and shifts, I think that it is better to be explicit what dunder customizes what operator.

While we are here, I think it is worth to document also the dunders for the `@` operator.

paulofreitas committed:
Improved shifting operators and added `@` operators

As per review request, improved shifting operators wording and added `@` operators.  👍

serhiy-storchaka commented on Doc/reference/expressions.rst:
```diff
@@ -1314,8 +1318,9 @@ integer; the result is that of mathematical division with the 'floor' function
 
-This operation can be customized using the special :meth:`~object.__truediv__` and
-:meth:`~object.__floordiv__` methods.
+This operation can be customized using the special :meth:`~object.__truediv__`,
```
There are two different operations.

paulofreitas replied:
> There are two different operations.

Nice catch, I was forgetting this one. 👍

paulofreitas committed:
Merge branch 'main' into rsub-arithmetic-operation

serhiy-storchaka commented on Doc/reference/expressions.rst:
```diff
@@ -1388,8 +1395,11 @@ The shifting operations have lower priority than the arithmetic operations:
-:meth:`~object.__rshift__` methods.
+The left shift operation can be customized using the special :meth:`~object.__lshift__`
+and :meth:`~object.__rlshift__` methods.
+
```
Would not be better to merge these two paragraphs?
```suggestion
```

paulofreitas replied:
For sure, merged it and did the same formatting for division and floor division operations. 👍

paulofreitas committed:
Separate division and floor division

Separates division and floor division and remove spacing from shifting operations.

paulofreitas committed:
Merge branch 'main' into rsub-arithmetic-operation

serhiy-storchaka reviewed (approved):
LGTM.

Thank you for your contribution @paulofreitas.
